lib/MusicBrainz/Server/WebService/Serializer/JSON/2/Role/Relationships.pm

```diff
@@ -11,11 +11,16 @@ around serialize => sub
     my ($orig, $self, $entity, $inc, $opts, $toplevel) = @_;
     my $ret = $self->$orig($entity, $inc, $opts, $toplevel);
 
-    return $ret unless defined $inc && $inc->has_rels;
+    return $ret unless
+        defined $inc &&
+        $inc->has_rels &&
+        $entity->are_relationships_loaded;
 
     my @rels = map { serialize_entity($_, $inc, $opts) }
-        sort_by { $_->target_key . $_->link->type->name }
-            @{ $entity->relationships };
+        sort_by { $_->link->type->name }
+        sort_by { $_->link_order // 0 }
+        sort_by { $_->target_key }
```

Comment:
Alternatively, it would be possible to request stable sort via `use sort 'stable';` (http://perldoc.perl.org/sort.html), but I think concatenating keys is better.